kolibri/core/decorators.py

```diff
@@ -339,7 +339,8 @@ def cache_no_user_data(view_func):
     _response = local()
 
     def render_and_cache(response, cache_key):
-        response.render()
+        if hasattr(response, "render") and callable(response.render):
+            response.render()
         etag = hashlib.md5(
             kolibri_version.encode("utf-8") + str(response.content).encode("utf-8")
```

Comment:
For response objects without a render function, is content guaranteed to be defined? Or will it just be null?

I wonder if it might be better to not cache an etag in this case.

Reply:
For such a case, we can check if `content` is defined for response object or not, after checking for the `render` method. This way, we'll only cache the etag if both the attributes are defined for a particular response object.

Reply:
We can check if `response.content` is defined and not null and then cache the etag, else we can `return None` for render_and_cache function. Should I proceed with these changes?

Reply:
Yes, I think that seems right.